From 3730ae0a9f3b8ed371642f31947b22c9bb6bac25 Mon Sep 17 00:00:00 2001
From: Valentine Barshak <vbarshak@dev.rtsoft.ru>
Date: Mon, 14 Jan 2013 17:56:31 +0400
Subject: [PATCH 023/127] arm: tegra: wdt: Allow to set WDT options when
 nowayout is set

This allows to use WDIOC_SETOPTIONS ioctl
when CONFIG_WATCHDOG_NOWAYOUT is enabled.

Signed-off-by: Valentine Barshak <vbarshak@dev.rtsoft.ru>
---
 drivers/watchdog/tegra_wdt.c |    4 ----
 1 files changed, 0 insertions(+), 4 deletions(-)

diff --git a/drivers/watchdog/tegra_wdt.c b/drivers/watchdog/tegra_wdt.c
index dd6acef..560337f 100644
--- a/drivers/watchdog/tegra_wdt.c
+++ b/drivers/watchdog/tegra_wdt.c
@@ -290,7 +290,6 @@ static long tegra_wdt_ioctl(struct file *file, unsigned int cmd,
 		return put_user(wdt->timeout, (int __user *)arg);
 
 	case WDIOC_SETOPTIONS:
-#ifndef CONFIG_WATCHDOG_NOWAYOUT
 		if (get_user(option, (int __user *)arg))
 			return -EFAULT;
 		spin_lock(&lock);
@@ -308,9 +307,6 @@ static long tegra_wdt_ioctl(struct file *file, unsigned int cmd,
 		}
 		spin_unlock(&lock);
 		return 0;
-#else
-		return -EINVAL;
-#endif
 	}
 	return -ENOTTY;
 }
-- 
1.7.4.4

